xm-commons-logging-configurable/src/test/java/com/icthh/xm/commons/logging/configurable/LoggingRefreshableConfigurationUnitTest.java

```diff
@@ -0,0 +1,302 @@
+
+    @MockBean
+    private TenantContextHolder tenantContextHolder;
+
+    @Mock
+    private TenantContext tenantContext;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        TestAppender.clearEvents();
+        when(tenantContextHolder.getContext()).thenReturn(new TenantContext() {
+            @Override
+            public boolean isInitialized() {
+                return true;
+            }
+
+            @Override
+            public Optional<Tenant> getTenant() {
+                return of(new PlainTenant(TenantKey.valueOf("TEST")));
+            }
+        });
+        when(tenantContext.getTenantKey()).thenReturn(Optional.of(TenantKey.valueOf("TEST")));
+
+        lepManager.beginThreadContext(ctx -> {
+            ctx.setValue(THREAD_CONTEXT_KEY_TENANT_CONTEXT, tenantContext);
+        });
+    }
+
+
+    @Test
+    public void testGreetingShouldReturnDefaultMessageTest() throws Exception {
+
+        testService.testMethodSecond("firstArgValue", "secondArtValue");
+        loggingRefreshableConfiguration.onRefresh(UPDATE_KEY, readConfig(TEST_LOG_UPDATE_CONFIG));
+
+    }
+
+    @Test
+    public void testUpdateConfiguration() {
+        loggingRefreshableConfiguration.onRefresh(UPDATE_KEY, readConfig(TEST_LOG_UPDATE_CONFIG));
+
+        //service
+        LogConfiguration serviceLoggingConfig =
+            loggingRefreshableConfiguration.getServiceLoggingConfig(null, "TestService", "testMethodFirst");
+        assertNotNull(serviceLoggingConfig);
+
+        //api
+        LogConfiguration apiLoggingConfig =
+            loggingRefreshableConfiguration.getApiLoggingConfig(null, "TestResource", "testMethodFirst");
+        assertNotNull(apiLoggingConfig);
+
+        //lep
+        LepLogConfiguration lepLoggingConfig =
+            loggingRefreshableConfiguration.getLepLoggingConfig("lep://TEST/general/TestLep$$default.groovy");
+        assertNotNull(lepLoggingConfig);
+    }
+
+    @Test
+    public void testChangeLogLevel() {
+        loggingRefreshableConfiguration.onRefresh(UPDATE_KEY, readConfig(TEST_LOG_UPDATE_CONFIG));
+
+        // service
+        testService.testMethodFirst("firstArgValue", "secondArgValue");
+        Optional<ILoggingEvent> logStart = TestAppender.getEvents().stream()
+            .filter(iLoggingEvent -> iLoggingEvent.getMessage().equals("srv:start: {}, input: {}"))
+            .findFirst();
+        assertTrue(logStart.isPresent());
+        assertEquals(logStart.get().getLevel(), Level.WARN);
+
+        Optional<ILoggingEvent> logStop = TestAppender.getEvents().stream()
+            .filter(iLoggingEvent -> iLoggingEvent.getMessage().equals("srv:stop:  {}, result: {}, time = {} ms"))
+            .findFirst();
+        assertTrue(logStop.isPresent());
+        assertEquals(logStop.get().getLevel(), Level.WARN);
+
+        //api
+        testResource.testMethodFirst("firstArgValue", "secondArgValue");
+        logStart = TestAppender.getEvents().stream()
```

Comment:
seems we can create method in TestAppender to make code more readable: 
```
public ILoggingEvent searchByMessage(String message){
events.stream()
            .filter(iLoggingEvent -> iLoggingEvent.getMessage().equals(messgae))
            .findFirst()
            .orElseThrow(() -> new RunTimeException("can not find message: "+message))
}
```

Reply:
fixed, added searchByMessage method